src/java/org/apache/cassandra/service/StorageProxy.java

```diff
@@ -2455,6 +2466,17 @@ protected void runMayThrow() throws Exception
         abstract protected void runMayThrow() throws Exception;
     }
 
+    public static void logRequestException(Exception exception, Collection<? extends ReadCommand> commands)
+    {
+        NoSpamLogger.log(logger, NoSpamLogger.Level.INFO, FAILURE_LOGGING_INTERVAL_SECONDS, TimeUnit.SECONDS,
+                         REQUEST_FAIL_MESSAGE,
+                         () -> new Object[]
+                               {
+                                   exception.getMessage(),
+                                   commands.stream().map(ReadCommand::toCQLString).collect(Collectors.joining("; "))
```

Comment:
nit; `";\n"`?

Reply:
The typical output looks something like this:

```
INFO  [node1_isolatedExecutor:1] node1 2021-11-18 10:05:26,416 NoSpamLogger.java:105 - "Cannot achieve consistency level ALL" while executing SELECT * FROM distributed_test_keyspace.tbl ALLOW FILTERING
```

With multiple commands...

```
INFO  [node1_isolatedExecutor:1] node1 2021-11-18 10:05:26,416 NoSpamLogger.java:105 - "Cannot achieve consistency level ALL" while executing SELECT * FROM distributed_test_keyspace.tbl WHERE id = 55 ALLOW FILTERING; SELECT * FROM distributed_test_keyspace.tbl WHERE id = 56 ALLOW FILTERING; SELECT * FROM distributed_test_keyspace.tbl WHERE id = 57 ALLOW FILTERING
```

I really have no preference in terms of how we separate them, given it shouldn't be that common to have a group anyway. I'll change it if you and @yifan-c are in agreement.

Reply:
I am ok keeping this